content/en/articles/js/react/unit-testing.md

```diff
@@ -0,0 +1,212 @@
+---
+date: "2015-11-24"
+title: Unit testing React components
+tags:
+  - js
+  - react
+  - tests
+authors:
+  - MoOx
+header:
+  image: https://farm6.staticflickr.com/5159/7112879347_9e0da289ca_z.jpg
+  credit: https://www.flickr.com/photos/dvids/7112879347
+  linearGradient: 160deg, rgba(0,0,0, .9), rgba(97, 218, 251, .7)
+---
+
+The React team has implemented a feature called
+[shallow rendering](http://facebook.github.io/react/docs/test-utils.html#shallow-rendering),
+which
+
+>lets you render a component "one level deep" and assert facts about
+what its render method returns, without worrying about the behavior of child
+components, which are not instantiated or rendered.
+This does not require a DOM.
+
+Sounds good, right? And guess what, shallow rendering is currently the
+[preferred way to test your React components](https://discuss.reactjs.org/t/whats-the-prefered-way-to-test-react-js-components/26).
+
+As you can see in the post mentioned at the end of this one, the
+actual code to test some components might seem a bit longer that what you could
+expect.
+
+Hopefully, [someone](https://github.com/vvo) created something pretty cool:
+[react-element-to-jsx-string](https://github.com/algolia/react-element-to-jsx-string).
+As the name of the package says, this library helps to render a react component
+into a JSX string.
+
+Now things start to become interesting:
+with those two things in mind (shallow render and react components as JSX strings),
+we can easily add some basic unit tests to some components.
+
+_There are others techniques to test React components, and most of those
+involve the DOM. This means you will need to run your tests in the browser (or
+using jsdom): your tests will be slower than the following method
+(which is more real unit testing since you execute less code and do not require
+a huge environment)._
+
+## Easy unit testing React components (without a DOM)
+
+Let's do this with the following (dump) component:
+
+```js
+// web_modules/Picture/index.js
+
+import React from "react"
+import { PropTypes } from "react"
+
+const Component = ({
```

Comment:
Not sure. Using something dumber will not be a realistic component that you might write in the real world. Don't you think?